File: Codes/Garnetwzy/225.cpp
class MyStack {
public:
    /** Initialize your data structure here. */
    queue<int> q;
    MyStack() {
        
    }
    
    /** Push element x onto stack. */
    void push(int x) {
        int m = q.size();
        q.push(x);
        for(int i = 0; i < m; i++) {
            q.push(q.front());
            q.pop();
        }
    }
    
    /** Removes the element on top of the stack and returns that element. */
    int pop() {
        int m = q.front();
        q.pop();
        return m;
    }
    
    /** Get the top element. */
    int top() {
        return q.front();
    }
    
    /** Returns whether the stack is empty. */
    bool empty() {
        return q.size() == 0;
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: codes/HellDream/225.java
class MyStack {
    private Queue<Integer> q1;
    private Queue<Integer> q2;

    public MyStack() {
        q1 = new ArrayDeque<>();
        q2 = new ArrayDeque<>();
    }
    /** Push element x onto stack. */
    public void push(int x) {
        if(q1.isEmpty()){
            q1.offer(x);
        }else{
            while(!q1.isEmpty()){
                int i = q1.poll();
                q2.offer(i);
            }
            q1.offer(x);
            while(!q2.isEmpty()){
                int i = q2.poll();
                q1.offer(i);
            }
        }
    }

    /** Removes the element on top of the stack and returns that element. */
    public int pop() {
        return q1.poll();
    }

    /** Get the top element. */
    public int top() {
        return q1.peek();
    }

    /** Returns whether the stack is empty. */
    public boolean empty() {
        return q1.isEmpty() &&q2.isEmpty();
    }
}
